examples/parallel_profiling.py

```diff
@@ -0,0 +1,164 @@
+        seconds.
+
+    """
+    gc.collect()
+    n_iter = 3
+    tic = time.time()
+    mem_use = [memory_usage((func, args, kwargs), interval=.00001,
+                            multiprocess=True, include_children=True)
+               for _ in range(n_iter)]
+    toc = time.time()
+    return (np.mean([max(mem_use_it) - min(mem_use_it)
+                     for mem_use_it in mem_use]),
+            (toc - tic) / n_iter)
+
+
+def print_info(mem_used_avg, elapsed_time):
+    print('Memory used by cropping the signal in the parent'
+          ' function: {:.2f} MiB'.format(mem_used_avg))
+    print('Elasped processing time: {:.2f} s\n'.format(elapsed_time))
+
+
+###############################################################################
+# As explained in the introduction, we will create a large array
+# (i.e. `signal`) and take is average at different locations (i.e. `slices`)
+# using a specific window size (i.e. `window_size`). To speed-up the execution,
+# we under-sample the number of slices to evaluate.
+
+import numpy as np
+
+signal = np.random.random((10000000,))
+window_size = 500000
+slices = [slice(start, start + window_size)
+          for start in range(0, signal.size - window_size, 50000)]
+
+
+###############################################################################
+# In the first strategy, the function passed to :class:`joblib.Parallel` will
+# take the portion of the signal from which we need to compute the
+# sum. Therefore, the function passed to parallel only compute the sum of the
+# array passed as argument. Therefore, all arrays passed to the
+# `moving_average` function will always be different and it will not be able to
+# share memory between the workers.
+
+from joblib import Parallel, delayed
+
+
+def process_cropping(signal, window_size, slices):
+    def moving_average(signal):
+        return signal.sum()
+
+    Parallel(n_jobs=2, max_nbytes=0)(delayed(
+        moving_average)(signal[sl]) for sl in slices)
+
+
+mem_used, elapsed_time = time_memory_profiling(process_cropping, signal,
+                                               window_size, slices)
+print_info(mem_used, elapsed_time)
+
+
+###############################################################################
+# The second strategy differs from the former since the original large array
+# will be passed to the `moving_average` function and the slicing will occur
+# directly in this function. Therefore, the original array can be shared
+# between the different workers and will lead to a more efficient use of the
+# memory.
+
+
+def process_slicing(signal, window_size, slices):
+    def moving_average(signal, sl):
+        return signal[sl].sum()
+
+    Parallel(n_jobs=2, max_nbytes=0)(delayed(
+        moving_average)(signal, sl) for sl in slices)
+
+
+mem_used, elapsed_time = time_memory_profiling(process_slicing, signal,
+                                               window_size, slices)
+print_info(mem_used, elapsed_time)
```

Comment:
Do I read it correctly that in this case it takes 30s instead of 4s and the memory usage is not that different 185 vs 234 MiB.

It seems like this is not a great behaviour of auto-memmapping (which is the default). I am not sure whether something is wrong or whether we can find a good use case for the auto-memmapping.